pennylane/devices/default_mixed.py

```diff
@@ -542,7 +589,15 @@ def _apply_operation(self, operation):
         if operation in diagonal_in_z_basis:
             self._apply_diagonal_unitary(matrices, wires)
         else:
-            self._apply_channel(matrices, wires)
+            num_op_wires = len(wires)
+            interface = qml.math.get_interface(*matrices)
+            print(interface)
+            # Use tensordot for Autograd and Numpy if there are more than 2 wires
+            # Use tensordot in any case for more than 7 wires, as einsum does not support this case
+            if (num_op_wires > 2 and interface in {"autograd", None}) or num_op_wires > 7:
+                self._apply_channel_tensordot(matrices, wires)
```

Comment:
An integration test is needed here

Reply:
Added decision based on interface of device state as well, and corresponding integration test sets :)